Foosball/FoosballApp/MainActivity.cs

```diff
@@ -0,0 +1,125 @@
+                    var record = FindViewById<Button>(Resource.Id.Record);
+                    var stop = FindViewById<Button>(Resource.Id.Stop);
+                    var play = FindViewById<Button>(Resource.Id.Play);
+                    var video = FindViewById<VideoView>(Resource.Id.VideoView);
+                    var backGame = FindViewById<Button>(Resource.Id.GameRecordBack); ;
+                    backGame.Click += delegate
+                    {
+                        SetContentView(Resource.Layout.QuickMatch);
+                    };
+                    record.Click += delegate
+                    {
+                        video.StopPlayback();
+                        recorder = new MediaRecorder();
+                        recorder.SetVideoSource(VideoSource.Default);
+                        recorder.SetAudioSource(AudioSource.Default);
+                        recorder.SetOutputFormat(OutputFormat.Default);
+                        recorder.SetVideoEncoder(VideoEncoder.Default);
+                        recorder.SetAudioEncoder(AudioEncoder.Default);
+                        recorder.SetOutputFile(path);
+                        recorder.SetPreviewDisplay(video.Holder.Surface);
+                        //recorder.Prepare();
+                        //recorder.Start();
+                    };
+                    stop.Click += delegate
+                    {
+                        if (recorder != null)
+                        {
+                            video.StopPlayback();
+                            recorder.Stop();
+                            recorder.Release();
+                            //this needs to be done
+                            //Proccesses proccesses = new Proccesses();
+                            //proccesses.ProccessVideo(pathtovideo);
+                            //proccesses.UploadResults(endpoint);
+                            //DeleteFile(path);
+                        }
+                    };
+                    play.Click += delegate
+                    {
+                        if (recorder == null)
+                        {
+                            //make it process the vid
+                        }
+                    };
+                };
+            };
+            doubl.Click += (s, arg) =>
+            {
+                SetContentView(Resource.Layout.TournamentMode);
+            };
+            troph.Click += (s, arg) =>
+            {
+                SetContentView(Resource.Layout.Statistics);
+            };
+            webe.Click += (s, arg) =>
+            {
+                var uri = Android.Net.Uri.Parse("http://localhost:4860/");
+                var intent = new Intent(Intent.ActionView, uri: uri);
+                StartActivity(intent);
+            };
+        }
+        protected override void OnDestroy()
+        {
+            base.OnDestroy();
+            if (recorder!= null)
+            {
+                recorder.Release();
```

Comment:
recorder?.Release() checks if recorder is null and if it is executes Release method. You can use that in this place